operations:nhc: document new option (#66)
omnivector-solutions/osd-documentation: operations/nhc.rst

        $ juju run-action slurmd/leader show-nhc-config --wait

NHC options
-----------

The default settings used in ``slurm.conf`` for NHC are as follows:

::

   HealthCheckProgram=/usr/sbin/omni-nhc-wrapper
   HealthCheckInterval=600
   HealthCheckNodeState=ANY,CYCLE

These values implies that NHC will run at every 600 seconds (10 minutes), on
all compute nodes regardless of their state (even on allocated nodes), but it
will not run on all of them at the same time.

> The script ``/usr/sbin/omin-nhc-wrapper`` allows you to supply custom arguments

Reviewer: ```suggestion
The ``/usr/sbin/omin-nhc-wrapper`` script allows you to supply custom arguments
```

to change how Slurm invokes the Health Check scripts via a ``charm-slurmctld``

> configuration. For example, for NHC to send an e-mail to ``admin@company.com``
> with the subject header ``NHC errors`` when it detects an error:

Reviewer: ```suggestion
configuration. For example, to configure NHC to send an e-mail to ``admin@company.com``
with the subject header ``NHC errors`` when it detects an error, supply the `health-check-params` argument as:
```

   $ juju config slurmd health-check-interval=300
   $ juju config slurmd health-check-state="CYCLE,ANY"

.. note::

>    NHC does not *undrain* a node. If a node was drained and NHC runs on that
>    node, the node will continue on the drained state, regardless of the checks
>    passing or failing.

Reviewer: ```suggestion
   NHC does not *undrain* a node. If NHC checks a drained node that no longer has disk issues,
   the node will remain in the drained state, regardless of the checks passing or failing.
```

Author: Hmmm I'm not sure about this... NHC goes way beyond disk issues.. I'll come back at this tomorrow morning.

>    This ensures that if someone drained a node for troubleshooting, this node
>    will not be resumed before the administrator finishes their tasks

Reviewer: ```suggestion
   This ensures that if someone drained a node for troubleshooting, it
   will not be resumed before the administrator finishes their tasks
```
